sympy/core/sympify.py

```diff
@@ -256,6 +256,20 @@ def sympify(a, locals=None, convert_xor=True, strict=False, rational=False,
     >>> kernS(s)
     -2*(-(-x + 1/x)/(x*(x - 1/x)**2) - 1/(x*(x - 1/x))) - 1
 
+    Floating point literals ... TODO: explain this (!)  Consider:
+
+    >>> sympify('0.1*1.4142135623730950488016887242097')**2
+    0.020000000000000002220446049250313
```

Comment:
I think your original example of `2**0.5` is better. The behavior in this case is basically correct (a product of floats should use the minimum of the precisions of the arguments), and it seems odd that you would want to change it.